### ISSUE TITLE
[Feature]: Update Components to Vue 3.5 Reactive Props Destructure

### ISSUE THREAD

**Issue by sadeghbarati** [radix-vue/shadcn-vue#742]
### Describe the feature

```diff
<script setup lang="ts">
import { type HTMLAttributes, computed } from 'vue'
import { AccordionItem, type AccordionItemProps, useForwardProps } from 'radix-vue'
import { cn } from '@/lib/utils'

- const props = defineProps<AccordionItemProps & { class?: HTMLAttributes['class'] }>()
+ const { class: className, ...props } = defineProps<AccordionItemProps & { class?: HTMLAttributes['class'] }>()

- const delegatedProps = computed(() => {
-  const { class: _, ...delegated } = props
- return delegated
- })


- const forwardedProps = useForwardProps(delegatedProps)
+ const forwardedProps = useForwardProps(props)
</script>

<template>
  <AccordionItem
    v-bind="forwardedProps"
-    :class="cn('border-b', props.class)"
+    :class="cn('border-b', className)"
  >
    <slot />
  </AccordionItem>
</template>
```

### Additional information

- [ ] I intend to submit a PR for this feature.
- [ ] I have already implemented and/or tested this feature.

**Comment by sadeghbarati** [comment]
I really don't like to reassign `class: className` 🙅 

But If I use only `class` I'll get this error 

<img width="1218" alt="Screenshot 2024-08-30 at 10 51 27 PM" src="https://github.com/user-attachments/assets/79cad6b4-ad73-4857-a929-20e0b92583dc">

---

~~@zernonia is this intended? should we ask it in `Vue Language Tools` repo about this?~~
Sorry this is eslint error, but how to remove this eslint rule?
It has runtime error check next Stackblitz please (vue-ts starter template)

**Comment by sadeghbarati** [comment]
<img width="1602" alt="Screenshot 2024-08-30 at 11 52 44 PM" src="https://github.com/user-attachments/assets/44e0c9ad-eb05-454c-8f15-ea732382e399">


https://stackblitz.com/edit/vitejs-vite-6eb22f?file=vite.config.ts,src%2Fcomponents%2FHelloWorld.vue&terminal=dev

Is this a reason why we have something like `className`? 😶‍🌫️ 


### How we can use `destructured class prop` without reassign it to `className` or any other names

**Comment by hrynevychroman** [comment]
@sadeghbarati I think one working approach will be `{ class: cl }` or something like this. This is not an ESLint error, this is a JavaScript (TS) preserved variable name:

How can we omit this issue:
<img width="953" alt="image" src="https://github.com/user-attachments/assets/d9077870-7100-4d55-9200-d59c5ea1d5c5">

Default error:
<img width="681" alt="image" src="https://github.com/user-attachments/assets/1613353d-ef17-40db-881e-83442f804af0">

Reserved words documentation link: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#reserved_words

<img width="751" alt="image" src="https://github.com/user-attachments/assets/4227dd6e-d231-4f07-a9d8-c2d357155ca2">

**Comment by kikky7** [comment]
Yeah, you can't name your variable class, that's why className is used. Vue props have different scope so you can name it class in there.

I usually name it cls or _class or something like that, className is too long and too reacty for me, but anyway you need to name it somehow, if you use such approach, because class is reserved word in JS.

**Comment by Saeid-Za** [comment]
I support the proposed change, but it is important to note that this is a breaking change. Anyone using Vue versions earlier than 3.5 without enabling the experimental flag may encounter unexpected errors. 

In my experience, there may not be an explicit error message, but the reactivity will fail to function properly, making it difficult to debug. 

If this is to be implemented, we would need to properly communicate the required migration steps.

**Comment by Dino-Kupinic** [comment]
Like @Saeid-Za said, this is a breaking change and imo most people would rather appreciate bug fixes and more mirroring with the official shadcn (new charts)
